Index: ExtensibleTreeView.cs
===================================================================
--- ExtensibleTreeView.cs	(revisión: 137515)
+++ ExtensibleTreeView.cs	(copia de trabajo)
@@ -92,7 +92,8 @@
 		private static Gtk.TargetEntry [] target_table = new Gtk.TargetEntry [] {
 			new Gtk.TargetEntry ("text/uri-list", 0, 11 ),
 			new Gtk.TargetEntry ("text/plain", 0, 22),
-			new Gtk.TargetEntry ("application/x-rootwindow-drop", 0, 33)
+			new Gtk.TargetEntry ("application/x-rootwindow-drop", 0, 33),
+			new Gtk.TargetEntry ("table/relationship", 0, 7777)	//Xiul change this but should be fixed later
 		};
 		
 		public IDictionary<string,string> ContextMenuTypeNameAliases {
